Mathlib/Analysis/SpecialFunctions/Pochhammer.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2025 Mitchell Horner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mitchell Horner
+-/
+import Mathlib.Analysis.Convex.Deriv
+import Mathlib.Data.Nat.Choose.Cast
+
+/-!
+# Pochhammer polynomials
+
+This file proves analysis theorems for Pochhammer polynomials.
+
+## Main definitions
+
+* `Differentiable.descPochhammer_eval` is the proof that the descending Pochhammer polynomial
+  `descPochhammer ℝ n` is differentiable.
+
+* `ConvexOn.descPochhammer_eval` is the proof that the descending Pochhammer polynomial
+  `descPochhammer ℝ n` is convex on `[n-1, ∞)`.
+
+* `descPochhammer_eval_le_sum_descFactorial` is a special case of **Jensen's inequality**
+  for `Nat.descFactorial`.
+
+* `descPochhammer_eval_div_factorial_le_sum_choose` is a special case of **Jensen's inequality**
+  for `Nat.choose`.
+-/
+
+
+section DescPochhammer
+
+variable {n : ℕ} {𝕜 : Type*} {k : 𝕜} [NontriviallyNormedField 𝕜]
+
+lemma descPochhammer_eval_eq_prod_range (n : ℕ) (k : 𝕜) :
+    (descPochhammer 𝕜 n).eval k = ∏ j ∈ Finset.range n, (k-j) := by
+  induction' n with n ih
```

Comment:
Throughout: my personal recommendation is that new code uses `induction` rather than `induction'`